internal/cli/atlas/deployments/setup.go

```diff
@@ -730,10 +734,62 @@ func (opts *SetupOpts) Run(ctx context.Context) error {
 	return opts.runConnectWith(cs)
 }
 
+func (opts *SetupOpts) RunAtlas(ctx context.Context) error {
+	s := setup.Builder()
+
+	// remove global flags and unknown flags
+	var newArgs []string
+	_, _ = log.Debugf("Removing flags and args from original args %s\n", os.Args)
+
+	flagstoRemove := map[string]string{
+		flag.TypeFlag:    "1",
+		flag.MDBVersion:  "1", // TODO: CLOUDP-200331
+		flag.ConnectWith: "1", // TODO: CLOUDP-199422
+	}
+
+	newArgs, err := workflows.RemoveFlagsAndArgs(flagstoRemove, map[string]bool{opts.DeploymentName: true}, os.Args)
+	if err != nil {
+		return err
+	}
+
+	// replace deployment name with cluster name
+	if opts.DeploymentName != "" {
+		newArgs = append(newArgs, fmt.Sprintf("--%s", flag.ClusterName), opts.DeploymentName)
+	}
+
+	// remove atlas deployments from args
+	newArgs = newArgs[2:]
+	// update args
+	s.SetArgs(newArgs)
```

Comment:
[nit] should we move this inside `workflows.RemoveFlagsAndArgs`?

Reply:
i'll move the first item, but i think we should set args for the command outside for now